### PR TITLE
DGE-94-Cryptographic-Key-Derivation

### PR DESCRIPTION
**Pull request by interc0der**
Bounty pull request for 0031 - DGE-94-Cryptographic-Key-Derivation

### CODE REVIEW

**Comment by jonathanlei** [comment]
Good start of implementation using existing xrpl-js tools, could you add some tests?

**Comment by JST5000** [comment]
For context this is referring to the xrpl.js code snippets bounty: https://github.com/XRPLBounties/Proposals/blob/main/bounties/0031%20JavaScript%20Code%20Samples.md

File: content/_code-samples/key-derivation/js/derive.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+'use strict'
+// Dependencies for Node.js.
+const api = require('ripple-keypairs')
+
+const entropy= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
```
Comment (JST5000) [inline content/_code-samples/key-derivation/js/derive.js:5]: nit:
```suggestion
const entropy = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
```

**Comment by JST5000** [comment]
For this bounty the goal is to have feature parity with these snippets so they can be used interchangeably depending on what language the new reader is most comfortable with: https://github.com/XRPLF/xrpl-dev-portal/tree/master/content/_code-samples/key-derivation/py

And specifically, the snippets are supposed to give a concrete way to understand these high-ish level concepts: https://xrpl.org/cryptographic-keys.html#key-derivation

So instead of using the client library to abstract away the details of the encode/decode, it's actually important to step through it and add comments explaining the intent. (The python one has some of those comments, and the high level concept also may have some useful things to point out in the code)

**Review by JST5000 — CHANGES_REQUESTED** [review]
See above comment

**Comment by JST5000** [comment]
Hey @interc0der I'm closing this PR as we have accepted a different submission for this bounty (https://github.com/XRPLF/xrpl-dev-portal/pull/1507).

**Pull request closed without merging**